src/Catalyst.Core.Modules.Dfs/CoreApi/UnixFsApi.cs

```diff
@@ -0,0 +1,467 @@
+            AddFileOptions options = default,
+            CancellationToken cancel = default)
+        {
+            await using var stream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read);
+            return await AddAsync(stream, Path.GetFileName(path), options, cancel).ConfigureAwait(false);
+        }
+
+        public async Task<IFileSystemNode> AddTextAsync(string text,
+            AddFileOptions options = default,
+            CancellationToken cancel = default)
+        {
+            await using var ms = new MemoryStream(Encoding.UTF8.GetBytes(text), false);
+            return await AddAsync(ms, "", options, cancel).ConfigureAwait(false);
+        }
+
+        public async Task<IFileSystemNode> AddAsync(Stream stream,
+            string name,
+            AddFileOptions options,
+            CancellationToken cancel)
+        {
+            options ??= new AddFileOptions();
+
+            // TODO: various options
+            if (options.Trickle)
+            {
+                throw new NotImplementedException("Trickle");
+            }
+
+            var blockService = GetBlockService(options);
+
+            var chunker = new SizeChunker();
+            var nodes = await chunker.ChunkAsync(stream, name, options, blockService, _keyApi, cancel)
+               .ConfigureAwait(false);
+
+            // Multiple nodes for the file?
+            var node = await BuildTreeAsync(nodes, options, cancel);
+
+            // Wrap in directory?
+            if (options.Wrap)
+            {
+                var link = node.ToLink(name);
+                var wlinks = new IFileSystemLink[]
+                {
+                    link
+                };
+                node = await CreateDirectoryAsync(wlinks, options, cancel).ConfigureAwait(false);
+            }
+            else
+            {
+                node.Name = name;
+            }
+
+            // Advertise the root node.
+            if (options.Pin && _dfsState.IsStarted)
+            {
+                await _dhtApi.ProvideAsync(node.Id, advertise: true, cancel: cancel).ConfigureAwait(false);
+            }
+
+            // Return the file system node.
+            return node;
+        }
+
+        private async Task<UnixFsNode> BuildTreeAsync(IEnumerable<UnixFsNode> nodes,
+            AddFileOptions options,
+            CancellationToken cancel)
+        {
+            var fsNodes = nodes as UnixFsNode[] ?? nodes.ToArray();
+            if (fsNodes.Length == 1)
+            {
+                return fsNodes.First();
+            }
+
+            // Bundle DefaultLinksPerBlock links into a block.
+            var tree = new List<UnixFsNode>();
+            for (var i = 0; true; ++i)
+            {
+                var bundle = fsNodes.Skip(DefaultLinksPerBlock * i).Take(DefaultLinksPerBlock);
+                var unixFsNodes = bundle.ToList();
+                if (!unixFsNodes.Any())
+                {
+                    break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)